torch/csrc/api/include/torch/nn/parallel/data_parallel.h

```diff
@@ -271,6 +271,10 @@ Tensor data_parallel(
 
   autograd::Scatter scatter(*devices, /*chunk_sizes=*/nullopt, dim);
   auto scattered_inputs = fmap<Tensor>(scatter.apply({std::move(input)}));
+  // Input tensor might not be big enough to scale across all available devices
+  if (scattered_inputs.size() < devices->size()) {
+    devices->resize(scattered_inputs.size(), Device(DeviceType::COMPILE_TIME_MAX_DEVICE_TYPES));
```

Comment:
Looks like the `Device(DeviceType::COMPILE_TIME_MAX_DEVICE_TYPES)` are not really used since this is shrinking?

Reply:
Correct, alternatively one can do something like:
` device = std::vector<Device>(devices->begin(), devices->begin() + scatterd_inputs.size());`
but, this is presumably tad slower than `resize()`